### PR TITLE
PARQUET-639: Do not export DCHECK in public headers

### PR DESCRIPTION
**Pull request by wesm**
I added a test so that DCHECK does not leak in the public headers. I prefer this to renaming the macro

### CODE REVIEW

File: src/parquet/public-api-test.cc [hunk 1]
```diff
@@ -29,4 +30,10 @@ TEST(TestPublicAPI, DoesNotIncludeThrift) {
 #endif
 }
 
+TEST(TestPublicAPI, DoesNotExportDCHECK) {
+#ifdef DCHECK
+  FAIL() << "parquet/util/logging.h should not transitively included";
```
Comment (majetideepak) [inline src/parquet/public-api-test.cc]: extreme nitpicking: "should not be transitively included"

**Comment by majetideepak** [comment]
+1 LGTM

**Comment by wesm** [comment]
Thanks. +1

**Comment by wesm** [comment]
Rebased. Waiting for green build

**Pull request closed without merging**